### DISCUSSION TITLE
Nested prefix view in /ipam/prefixes

### DISCUSSION THREAD

**Discussion by Alph4d0g** [netbox-community/netbox#8774] [Q&A]
Hi folks, I've googled for the last 3 hours on this, unfortunately without success.

I have a prefix setup as a container in VRF Global, with prefixes below in different VRFs. Clicking on the "Child Prefixes" tab, I can see them nicely nested like this:
- 192.168.0.0/16           //VRF Global
   - 192.168.0.0/19        //VRF A
      - 192.168.15.0/24   //VRF A
   - 192.168.32.0/19      //VRF B
   - 192.168.128.0/17    //VRF B
   - 192.168.255.0/24    //VRF Transit_A
   - 192.168.255.0/24    //VRF Transit_B
   - 192.168.255.0/24    //VRF Transit_C

However, in the Global "prefix" section (https://yournetboxhost/ipam/prefixes) it does not show ANY Parent-Child relation anymore. 
Also it displays 0 Child prefixes in the column. Does anyone know if this wanted behavior?

In get_child_prefixes(self) it seems to be automatically returning child prefixes belonging to any VRF. Is this NOT used in the table view?

2nd. if that is wanted behavior, Is it possible to create one's own views without touching the original code (to keep local deployments upgradeable)? 

> ```
>     def get_child_prefixes(self):
>         """
>         Return all Prefixes within this Prefix and VRF. If this Prefix is a container in the global table, return child
>         Prefixes belonging to any VRF.
>         """
> ```

One thing that can lead to even more confusion is the different information netbox provides with regards to number child prefixes.
In one view it tells there's 0 children, but if I click on it (e.g. /ipam/prefixes/258/), it tells me 15.

Currently, ntbx is running on v2.11.12, behavior stays the same in the official demo.

Best,
Sebastian

**Comment by candlerb** [comment]
Is the issue that when you're browsing a global prefix, you can't see VRF-assigned prefixes within it? I think that's normal. VRFs are effectively "network namespaces" in Netbox: independent sets of IP allocations which can overlap. So 192.168.0.0/19 (VRF A) is not a child of 192.168.0.0/16 (VRF Global); it is a child of 192.168.0.0/16 (VRF A).

**Reply by Alph4d0g** [reply]
Then why is there an inconsistency in the views (I get the concept and it's most likely the best solution)? When I browse the 192.168.0.0/16 prefix, it DOES show me child prefixes within different VRFs (which I personally want that way), but in the Global view it doesn't.

**Reply by candlerb** [reply]
Just for clarity, can you give the two browser URL paths for the two cases you are describing?

One is `/ipam/prefixes/` and one is `/ipam/prefixes/NNN/` ?

**Reply by Alph4d0g** [reply]
one is `/ipam/prefixes/`, the other is `/ipam/prefixes/258/prefixes/` with 258 being my container in the Global VRF
